app/concerns/agent_controller_concern.rb

```diff
@@ -68,7 +68,8 @@ def control!
             log "Agent '#{target.name}' is disabled"
           end
         when 'configure'
-          target.update! options: target.options.deep_merge(interpolated['configure_options'])
+          new_options = target.options.deep_merge(interpolated['configure_options']) { |key, old, new| !(new.kind_of?(Array) && old.kind_of?(Array)) ? new : (old+new).uniq }
```

Comment:
Maybe the CommanderAgent should have an `options_update_mode` of either `replace` (the current system) or `append` (your new code).  I agree that it still doesn't cover the case that you mentioned, ...but it's an improvement!